src/VisualStudio/Core/Def/Implementation/ProjectSystem/MiscellaneousFilesWorkspace.cs

```diff
@@ -328,15 +329,40 @@ private void AttachToDocument(uint docCookie, string moniker)
             // This should always succeed since we only got here if we already confirmed the moniker is acceptable
             var languageInformation = TryGetLanguageInformation(moniker);
             Contract.ThrowIfNull(languageInformation);
-            var parseOptions = languageInformation.ParseOptions;
 
-            if (Path.GetExtension(moniker) == languageInformation.ScriptExtension)
+            var languageServices = Services.GetLanguageServices(languageInformation.LanguageName);
+            var compilationFactory = languageServices.GetService<ICompilationFactoryService>();
+            var compilationOptions = compilationFactory.GetDefaultCompilationOptions();
+            var syntaxTreeFactory = languageServices.GetService<ISyntaxTreeFactoryService>();
+            var parseOptions = syntaxTreeFactory.GetDefaultParseOptions();
```

Comment:
We should check if TypeScript or F# or others are using this workspace. I'm unsure, but it looks terrifyingly possible that they could have allowed ParseOptions to be null in that case and now we'd be crashing for not having these services. @brettfo? @paulvanbrenk?